### ISSUE TITLE
3.65 breaks preview giving 'error: provided project name "" doesn't match Pulumi.yaml'

### ISSUE THREAD

**Issue by invertedtomato** [pulumi/pulumi#12760]
### What happened?

Our CI/CD pipeline runs the following:
```
pulumi stack select stg
pulumi preview
```

This worked file on `3.64.0` however on `3.65.0` fails with the following error:

`error: provided project name "" doesn't match Pulumi.yaml`

### Expected Behavior

Preview is generated

### Steps to reproduce

Run the following in our context on `3.65.0`
```
pulumi preview
```

Are there any other factors specific to our context?

### Output of `pulumi about`

```
CLI          
Version      3.65.0
Go Version   go1.20.3
Go Compiler  gc

Host     
OS       darwin
Version  13.2.1
Arch     arm64

Pulumi locates its logs in /var/folders/tb/h0ws1ttd72q8ssn96cp2ch7m0000gn/T/ by default
```

### Additional context

Same issue occurs for `pulumi up`

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by justinvp** [comment]
Confirmed:

```
$ pulumi login --local
$ pulumi new go -g
$ pulumi stack init dev
error: could not create stack: provided project name "" doesn't match Pulumi.yaml
```

Appears to be due to https://github.com/pulumi/pulumi/pull/12728

After reverting that commit, I no longer get the error. Going to open a PR with the revert and kick off a patch release.

**Comment by invertedtomato** [comment]
Confirmed fixed for me in 3.65.1. Thanks guys!
